src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE = 'none',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+type Props<T> = {
+  goods: T[];
+  sortGood: SortType;
+  isReversed: boolean;
+};
+
+function getPreparedGoods<T>({ goods, sortGood, isReversed }: Props<T>): T[] {
+  const preparedGoods = [...goods];
+
+  if (sortGood !== SortType.NONE) {
+    preparedGoods.sort((good1, good2) => {
+      let result: number;
+
+      switch (sortGood) {
+        case SortType.ALPHABET:
+          result = String(good1).localeCompare(String(good2));
+          break;
+
+        case SortType.LENGTH:
+          result = String(good1).length - String(good2).length;
+          break;
+
+        default:
+          result = 0;
+          break;
+      }
+
+      return result;
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortGood, setSortGood] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const preparatedGoods = getPreparedGoods({
```

Comment:
The variable `preparatedGoods` is misspelled. It should be `preparedGoods` to match the function name `getPreparedGoods` and maintain consistency.